core/src/kvs/lq_v2_fut.rs

```diff
@@ -0,0 +1,205 @@
+use crate::cf;
+use crate::cf::{ChangeSet, TableMutation};
+use crate::dbs::{Options, Statement};
+use crate::err::Error;
+use crate::fflags::FFLAGS;
+use crate::kvs::lq_cf::LiveQueryTracker;
+use crate::kvs::lq_structs::{LqIndexKey, LqIndexValue, LqSelector};
+use crate::kvs::LockType::Optimistic;
+use crate::kvs::TransactionType::Read;
+use crate::kvs::{construct_document, Datastore, Transaction};
+use crate::sql::statements::show::ShowSince;
+use crate::vs::conv;
+use futures::lock::Mutex;
+use std::collections::BTreeMap;
+use std::sync::Arc;
+use tokio::sync::RwLock;
+
+/// Poll change feeds for live query notifications
+pub async fn process_lq_notifications(ds: &Datastore, opt: &Options) -> Result<(), Error> {
+	// Runtime feature gate, as it is not production-ready
+	if !FFLAGS.change_feed_live_queries.enabled() {
+		return Ok(());
+	}
+	// Return if there are no live queries
+	if ds.notification_channel.is_none() {
+		trace!("Channels is none, short-circuiting");
+		return Ok(());
+	}
+	if ds.lq_cf_store.read().await.is_empty() {
+		// This is safe - just a shortcut
+		trace!("No live queries, short-circuiting");
+		return Ok(());
+	}
+
+	// Change map includes a mapping of selector to changesets, ordered by versionstamp
+	let mut relevant_changesets: BTreeMap<LqSelector, Vec<ChangeSet>> = BTreeMap::new();
+	{
+		let tx = ds.transaction(Read, Optimistic).await?;
+		populate_relevant_changesets(
+			tx,
+			ds.lq_cf_store.clone(),
+			ds.engine_options.live_query_catchup_size,
```

Comment:
Shall we just pass `ds` to gain clarity?

Reply:
I think for this one I would like to keep it. Reason being that the "units" required for the function are not "Datastore", only the LQ-CF store and the usize catchup size. It shouldn't know about Datastore.